kafka-connect/kafka-connect-events/src/main/java/org/apache/iceberg/connect/events/DataWritten.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.connect.events;
+
+import java.util.List;
+import java.util.UUID;
+import org.apache.avro.Schema;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.DeleteFile;
+import org.apache.iceberg.types.Types.ListType;
+import org.apache.iceberg.types.Types.NestedField;
+import org.apache.iceberg.types.Types.StructType;
+import org.apache.iceberg.types.Types.UUIDType;
+
+/**
+ * A control event payload for events sent by a worker that contains the table data that has been
+ * written and is ready to commit.
+ */
+public class DataWritten implements Payload {
+
+  private StructType partitionType;
+
+  private UUID commitId;
+  private TableReference tableReference;
+  private List<DataFile> dataFiles;
+  private List<DeleteFile> deleteFiles;
+  private StructType icebergSchema;
+  private Schema avroSchema;
+
+  // Used by Avro reflection to instantiate this class when reading events
+  public DataWritten(Schema avroSchema) {
+    this.avroSchema = avroSchema;
```

Comment:
The schema passed by Avro may be a projection of the full type. That's why in `GenericDataFile` and similar classes, we keep a mapping from this schema's fields to the canonical field positions. This should be fine as long as the schema passed in doesn't project columns (and I don't see why it would) but this is technically unsafe.

What about adding a precondition check that the schema matches?

Reply:
I think that would cause problems w/ schema evolution? e.g. if we add a new field to the event then we could no longer deserialize old events.

Reply:
I added mapping of position to field ID.